lms/djangoapps/courseware/views/views.py

```diff
@@ -928,7 +930,7 @@ def course_about(request, course_id):
 
         # Used to provide context to message to student if enrollment not allowed
         can_enroll = bool(request.user.has_perm(ENROLL_IN_COURSE, course))
-        invitation_only = course.invitation_only
+        invitation_only = is_course_default_invite_only_enabled() or course.invitation_only
```

Comment:
Will course.invitation_only always be True or False, or can it be not set (None)? 

It strikes me that with the logic here, if the toggle is set to True, it will never be possible to override it for one course. All courses will be invite only.

Reply:
@pdpinch It either be `True/False`. 

> It strikes me that with the logic here, if the toggle is set to True, it will never be possible to override it for one course. All courses will be invite only.

hmmm but what I get yet, We are giving precedence to `COURSE_DEFAULT_INVITE_ONLY` flag in the task. 
I mean, If `is_course_default_invite_only_enabled ` is `True` then all courses will be invite only. Is it not the requirement here ? mean, reference to earlier merge:  [PR](https://github.com/mitodl/edx-platform/pull/126)

Reply:
As is, the PR meets the MIT ODL use case, but I feel like the setting name suggests a different behavior. `COURSE_DEFAULT_INVITE_ONLY` sounds like it should only change the default setting, instead of overriding the setting for all courses. 

We could either:

- Change the logic, or
- Change the setting to `COURSES_INVITE_ONLY`